### PR TITLE
Add a sniff that enforces deprecated annotations to have a description

### PR DESCRIPTION
**Pull request by gmponos**
Hi,

As the title says.. Not sure if you are willing to include a sniff like this..

If you are.. besides the missing tests let me know what I can do to be acceptable

### CODE REVIEW

**Comment by kukulich** [comment]
Yes, I’m ok to merge it :)

File: SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\Commenting;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use const T_DOC_COMMENT_TAG;
+
+class DeprecatedDocRequiresDescriptionSniff implements Sniff
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php:9]: `DeprecatedAnnotation...` please.

File: SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+	 * @param File $phpcsFile
+	 * @param int $propertyPointer
+	 */
+	public function process(File $phpcsFile, $propertyPointer): void
+	{
+		$tokens = $phpcsFile->getTokens();
+
+		if ($tokens[$propertyPointer]['content'] !== '@deprecated') {
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php:29]: Use `AnnotationHelper` please. It uses better parser.

File: tests/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniffTest.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\Commenting;
+
+use SlevomatCodingStandard\Sniffs\TestCase;
+
+class DeprecatedDocRequiresDescriptionSniffTest extends TestCase
+{
+
+	public function testErrors(): void
```
Comment (kukulich) [inline tests/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniffTest.php:10]: Please split test to `testErrors` and `testNoErrors`.

File: tests/Sniffs/Commenting/data/deprecatedDocRequiresDescription.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * @deprecated
+ */
+class Whatever
+{
+	/**
+	 * @deprecated
+	 */
+	public const DEPRECATED_WITHOUT = 'deprecated';
```
Comment (kukulich) [inline tests/Sniffs/Commenting/data/deprecatedDocRequiresDescription.php:13]: Please use tabs in the whole file.

File: SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php (hunk 2 above)
Comment (gmponos) [reply SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php:29]: Sorry not familiar with this class.. but does it return the `deprecated`? https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Helpers/AnnotationHelper.php#L276

what's the object I should expect if I use the `getAnnotationsByName`?

File: tests/Sniffs/Commenting/data/deprecatedDocRequiresDescription.php (hunk 4 above)
Comment (vladyslavstartsev) [reply tests/Sniffs/Commenting/data/deprecatedDocRequiresDescription.php:13]: @kukulich you might want to move this check (tabs) to the CI level

**Comment by kukulich** [comment]
@gmponos Will you continue in this PR?

**Comment by gmponos** [comment]
@kukulich I was expecting an answer here: https://github.com/slevomat/coding-standard/pull/1044#discussion_r444493243

File: SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php (hunk 2 above)
Comment (kukulich) [reply SlevomatCodingStandard/Sniffs/Commenting/DeprecatedDocRequiresDescriptionSniff.php:29]: This class is part of this repository, so everything is in the code...

**Review by kukulich — APPROVED** [review]

**Pull request merged**